charms/istio-pilot/src/resources_handler.py

```diff
@@ -90,16 +90,16 @@ def delete_manifest(
                 ignore_unauthorized=ignore_unauthorized,
             )
 
-    def generate_generic_resource_class(self, filename: str) -> GenericNamespacedResource:
+    def get_custom_resource_class_from_filename(self, filename: str) -> GenericNamespacedResource:
```

Comment:
This can return a Global resource as well

Reply:
Yeh, is `Any` good enough?

Reply:
could also have a `Union[]`, but this is good